common/src/main/java/tachyon/retry/CountingRetry.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.retry;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * An option which allows retrying based on maximum count.
+ */
+public class CountingRetry implements RetryPolicy {
+
+  private final int mMaxRetries;
+  private int mCount = -1;
```

Comment:
The reason for the subtraction is to get the correct number of counts tried. My thought is mMaxRetries should always match the count of retries when it runs out of all attempts and as you had suggested before if we just do "return mCount ++ < mMaxRetries" in attemptRetry that value will be incorrect since it would always throw an incremented mCount in the getter 

For example, If mMaxReTries = 1, mCount will say 2 if we just have "return mCount ++ < mMaxRetries".  The latest updated code is consistent with other RetryPolicy and  the getter just returns mCount